HerPublicWebsite.BusinessLogic/ExternalServices/OsPlaces/OsPlacesDpaDto.cs

```diff
@@ -1,23 +1,86 @@
+using System.Globalization;
+using HerPublicWebsite.BusinessLogic.Models;
 using Newtonsoft.Json;
 
-public record class OsPlacesDpaDto
+namespace HerPublicWebsite.BusinessLogic.ExternalServices.OsPlaces;
+
+public class OsPlacesPostcodeResponseDto
 {
+    [JsonProperty("results")]
+    public List<OsPlacesPostcodeResultDto> Results { get; set; }
+}
+
+public class OsPlacesPostcodeResultDto
+{
+    [JsonProperty("DPA")]
+    public OsPlacesDpaDto Dpa { get; set; }
+}
 
+public class OsPlacesDpaDto
+{
+    private static readonly TextInfo GbTextInfo = new CultureInfo("en-GB").TextInfo;
+    public Address Parse()
+    {
+        return new Address()
+        {
+            AddressLine1 = $"{BuildingNumber ?? ToTitleCase(BuildingName)} {ToTitleCase(DependentThoroughFareName) ?? ToTitleCase(ThoroughFareName)}",

```

Comment:
I don't think this is robust enough - for example if I put in my old postcode (NW5 2HA), all the flats have the same building number, name, street and postcode:
![image](https://github.com/UKGovernmentBEIS/desnz-home-energy-retrofit-beta/assets/9695026/e2c7eacb-6da3-46a2-b819-60976170ad3c)

The address should read: **Flat 4, Lambton House, 22 Fortess Road, London, NW5 2HA**, with the differentiator being `SUB_BUILDING_NAME`.

I think we either need to include all the address fields, or just use the `ADDRESS` field on the OS Places response?

Reply:
Other slightly pathological examples:
- W2 1JB has 100 results
- CH5 3QW has a whole load of components
- IM9 4AJ has no street name

Reply:
Good find. The problem with using the ADDRESS field is that it's in all-caps which is a bit ugly, and if we change it to title case we need to identify and exclude the postcode from that.

Reply:
Yeah, I guess I'm a bit nervous we might miss more cases if we reimplement the address concatenation logic, but I agree it looks ugly!

Reply:
Added a fix for postcodes with more than 100 results

Reply:
Updated the address parsing so that all fields are included if they are present in the response.